manifests/init.pp

```diff
@@ -16,29 +16,49 @@
   $resolver_config_file_owner  = 'root',
   $resolver_config_file_group  = 'root',
   $resolver_config_file_mode   = '0644',
-) {
+  ){
 
-  # Validates domain
-  if is_domain_name($domain) != true {
-    fail("Domain name, ${domain}, is invalid.")
+  if $::kernel == 'windows' {
+    if is_array($nameservers) {
+      $stringifynameservers = join($nameservers,',')
+      $testserver = $nameservers[0]
+    }
+    else {
+      $stringifynameservers = $nameservers
+      $testserver = $nameservers
+    }
+    exec { 'set dns':
+      command  => "\$interface = find-netroute -RemoteIPAddress ${testserver} | select -expand 'InterfaceAlias'; set-dnsclientserveraddress -interfacealias \$interface -serveraddress {${stringifynameservers}}",
```

Comment:
You know of a way to validate powershell in Linux? Thinking that these could go in a template, so that the separate files could be validated.